cpp/src/io/csv/csv_writer.cu

```diff
@@ -45,9 +45,9 @@ NVStrings* column_to_strings_csv( gdf_column* column, const char* delimiter, con
         case GDF_STRING:
             rtn = (static_cast<NVStrings*>(column->data))->copy();
             break;
-        //case GDF_BOOL:
-        //    rtn = NVStrings::create_from_bools((const bool*)col->data,rows,true_string,false_string,valid);
-        //   break;
+        case GDF_BOOL8:
+            rtn = NVStrings::create_from_bools(static_cast<const bool*>(column->data),rows,true_string,false_string,valid);
```

Comment:
This isn't safe.

You're casting a column of `chars` to a column of `bool`s and those are never guaranteed to be the same thing.

Reply:
Should it be `static_cast<const cudf::bool8*>(column->data)` (will that properly call the `explicit operator bool` in `cudf::wrapper`?)

If not, then `static_cast<const bool*>(static_cast<const cudf::bool8*>(column->data))`...

Reply:
When I uncommented the line, I thought it wouldn't work but the output seems correct for different values I've tried in the gtest...

I would've expected the output to be completely wrong so I thought it's automagically converting it through the cast...

Reply:
The only solution is `NVStrings::create_from_bools` needs to be updated to work with `cudf::bool8` columns.

Individual elements of `cudf::bool8` can be casted to `bool` because of the `operator bool()`. However, any attempt to cast an _array_ of `cudf::bool8` to `bool` is very unsafe, because there is no guarantee that `sizeof(cudf::bool8) == sizeof(bool)`. For example, many C++ implementations specialize the implementation of `std::vector<bool>` to be 1 bit per `bool`.

Reply:
NvStrings itself doesn't use cudf types though...

I think the best way is for NvStrings to have a more generic API that takes in an array of variable type size, and a dictionary that will map type values to whatever string values.

We can open a feature request PR in nvstrings for that. For this PR, I'm using an intermediate buffer to use the current interface... Let me know if there's a better short-term solution...

Reply:
NvStrings is being merged into cuDF over time, so eventually NVStrings _will_ use cudf types.

Reply:
Copying to an intermediate `bool` column makes sense to me for an intermediate solution.

Reply:
> NvStrings is being merged into cuDF over time, so eventually NVStrings _will_ use cudf types.

Ah, didn't know that. That would be great!